template/cordova/lib/PluginInfo.js

```diff
@@ -0,0 +1,150 @@
+        var configFiles = parentGetConfigFiles(platform);
+        return processChanges(configFiles);
+    };
+
+    function processChanges(changes) {
+        var hasManifestChanges  = checkManifestChanges(changes);
+
+        // Demux 'package.appxmanifest' into relevant platform-specific appx manifests.
+        // Only spend the cycles if there are version-specific plugin settings
+        if (hasManifestChanges)
+        {
+            var oldChanges = changes;
+            changes = [];
+
+            oldChanges.forEach(function(change, changeIndex) {
+                // Only support semver/device-target demux for package.appxmanifest
+                // Pass through in case something downstream wants to use it
+                if (change.target !== 'package.appxmanifest') {
+                    changes.push(change);
+                    return;
+                }
+
+                // No semver/device-target for this config-file, pass it through
+                if (!(hasChangeVersion(change) || hasChangeTarget(change))) {
+                    // New windows template separate manifest files for Windows10, Windows8.1 and WP8.1
+                    changes = changes.concat(demuxChangeWithSubsts(change));
+                    return;
+                }
+
+                var targetDeviceSet = getDeviceTargetSetForChange(change);
+                changes = replaceChangeTarget(targetDeviceSet, change, changes);
+            });
+        }
+        return changes;
+    }
+
+    function checkManifestChanges(changes) {
+        return changes.some(function(change) {
+            return change.target === 'package.appxmanifest';
+        });
+    }
+    function hasChangeVersion(change) {
+        return (typeof change.versions !== 'undefined');
+    }
+
+    function hasChangeTarget(change) {
+        return (typeof change.deviceTarget !== 'undefined');
+    }
+
+    function demuxChangeWithSubsts(change) {
+        return SUBSTS.map(function(subst) {
+             return createChangeWithNewTarget(change, subst);
+        });
+    }
+
+    function createChangeWithNewTarget(change, substTarget) {
+        var changeWithNewTarget = {};
+        Object.keys(change).forEach(function(changeKey) {
+            changeWithNewTarget[changeKey] = change[changeKey];
+        });
+        changeWithNewTarget.target = substTarget;
+        return changeWithNewTarget;
+    }
+
+    function getDeviceTargetSetForChange(change) {
+        var targetDeviceSet = hasChangeTarget(change) ? change.deviceTarget : 'all';
+        if (TARGETS.indexOf(targetDeviceSet) === -1) {
+            // target-device couldn't be resolved, fix it up here to a valid value
+            targetDeviceSet = 'all';
+        }
+        return targetDeviceSet;
+    }
+
+    // This is a local function that creates the new replacement representing the
+    // mutation.  Used to save code further down.
+    function createReplacement(manifestFile, originalChange) {
```

Comment:
This looks very similar to `createChangeWithNewTarget` function - consider merging into one function